api/taskcluster/webhook.go

```diff
@@ -194,12 +204,40 @@ func shouldProcessStatus(log shared.Logger, processAllBranches bool, status *sta
 	return true
 }
 
-func extractTaskGroupID(targetURL string) string {
-	lastSlash := strings.LastIndex(targetURL, "/")
-	if lastSlash == -1 {
-		return ""
+// handleCheckRunEvent processes an Azure Pipelines check run "completed" event.
+func handleCheckRunEvent(aeAPI shared.AppEngineAPI, event *github.CheckRunEvent) (bool, error) {
```

Comment:
It might be less confusing to move this function into `api/taskcluster/api.go` (and perhaps even inline it into the impl), since nothing else is using this function and we can make this file specific for legacy Taskcluster webhooks.